python/pacemaker/_cts/corosync.py

```diff
@@ -0,0 +1,81 @@
+""" A module providing functions for manipulating corosync """
+
+__all__ = ["generate_corosync_cfg", "start_corosync", "stop_corosync"]
+__copyright__ = "Copyright 2009-2023 the Pacemaker project contributors"
+__license__ = "LGPLv2.1+"
+
+import os
+import time
+import subprocess
+
+from pacemaker.buildoptions import BuildOptions
+from pacemaker._cts.process import killall
+
+
+AUTOGEN_COROSYNC_TEMPLATE = """
+totem {
+    version: 2
+    cluster_name: %s
+    crypto_cipher: none
+    crypto_hash: none
+    transport: udp
+}
+
+nodelist {
+    node {
+        nodeid: 1
+        name: %s
+        ring0_addr: 127.0.0.1
+    }
+}
+
+logging {
+    debug: off
+    to_syslog: no
+    to_stderr: no
+    to_logfile: yes
+    logfile: %s
+}
+"""
+
+def corosync_cfg_exists():
+    """ Does the corosync config file exist? """
+
+    return not os.path.exists(BuildOptions.COROSYNC_CONFIG_FILE)
+
+def generate_corosync_cfg(logdir, cluster_name, node_name):
+    """ Generate the corosync config file, if it does not already exist """
+
+    if corosync_cfg_exists():
+        return
+
+    logname = os.path.join(logdir, "corosync.log")
+    with open(BuildOptions.COROSYNC_CONFIG_FILE, "w", encoding="utf-8") as corosync_cfg:
+        corosync_cfg.write(AUTOGEN_COROSYNC_TEMPLATE % (cluster_name, node_name, logname))
+
+def start_corosync(verbose):
+    """ Start the corosync process """
+
+    if verbose:
+        print("Starting corosync")
+
+    with subprocess.Popen("corosync", stdout=subprocess.PIPE) as test:
+        test.wait()
+
+    time.sleep(10)
```

Comment:
Might as well get this library started off right.  I'll work on this and add a patch on top.

Reply:
FYI the lab has some log parsing code already. Not sure if any of it will be usable here since it's based on ssh'ing in from the exerciser node and uses the lab's logging functions, but see watcher.py

Reply:
It doesn't look like we need anything especially clever to figure this out, though what we really need is code to grab the logfile from the corosync config file.  And that is pretty cheesy.  I'd be happy to consider alternate approaches.